### DISCUSSION TITLE
Actividad 4 - Grupo 10

### DISCUSSION THREAD

**Discussion by MilagrosSoberon** [frlp-utn-ingsoft/catedra#156] [Show and tell]
https://github.com/MilagrosSoberon/vetsoft

**Comment by ivanduva** [comment]
Buenas! Les puedo pedir el link a donde lo deployaron?

**Reply by ivanduva** [reply]
Les recuerdo nuevamente pasar el link al deploy

**Reply by nuriarobledo** [reply]
hola! mil disculpas no nos llegaban las notificaciones!! 
este es el link:  https://vetsoft-g10.onrender.com/

**Reply by ivanduva** [reply]
Buenas,
Les vamos a pedir algunas correcciones:

- En main la app se rompe al guardar con valores erróneos en el teléfono y hay diferencias entre develop y main. Debería quedar todo correcto en main. Tal vez no haya ayudado tener esa capa intermedia, revisen si es necesario tener develop o si les conviene sacarla.
- El form de editar no valida y permite guardar valores incorrectos. Por lo tanto tampoco están validando el endpoint y el modelo.
- En cuanto a la validación del nombre, les vamos a pedir un test e2e.
- En cuanto a la validación de que el teléfono sea numérico, no están haciendo la validación en el modelo. Agreguen eso y agreguen los tests de unidad e integración que prueben que aparezca el error. Los que agregaron son para el caso de que el teléfono sea vacío. También les pedimos que agreguen un e2e.
- En cuanto a la validación del mail, les pedimos que agreguen un test de unidad.
- En cuanto a la validación de que el teléfono comience con 54, les pedimos que agreguen un test e2e.

Recuerden que los cambios deben ser realizados por las personas que se encargaron de cada issue.

Después les dejamos algunos comentarios sobre procedimiento que deberían revisar en las próximas correcciones:

- Vimos que hicieron commits sobre developo main para agregar lo relativo a la CI/CD. Si estamos usando el workflow Feature Branch, esto es peligroso y constituye una muy mala práctica. Estos commits no pasan por una instancia formal de review y por lo tanto es poco seguro, puede romper la aplicación. Hay que hacer PRs para cumplir con el proceso.
- Vemos algunos malos mensajes de commit. En algunos casos se nombran archivos o funciones lo cual es redundante, en otros casos vimos mensajes de commit repetidos con respecto al anterior como este https://github.com/MilagrosSoberon/vetsoft/commit/16a7f0f1f566d88270d3761126cb3ae113489448. Si fue porque hicieron un merge y al solucionar conflictos volvieron a escribir el mismo mensaje, en ese caso recomendamos poner el mensaje por default de git. Recuerden que los mensajes de commit deberían ser una síntesis de lo que el commit soluciona en cuanto a funcionalidad.

Por otro lado, hay varios puntos altos como la documentación de los PRs y un buen trabajo en una parte de los tests.

Arreglando lo anterior ya tendrían el trabajo integrador completamente aprobado.

**La última fecha límite de será el último día de cursada antes de las vacaciones: Lunes 8 de Julio 23:59 hs.**

Saludos

**Reply by sebasgonzales** [reply]
Buenas! 
Ya hicimos los cambios necesarios y quedó todo funcionando. En cuanto a las diferencias entre main y develop, les pedimos disculpas porque fue por un malentendido nuestro, nos falto realizar un pr, pero ahora ya quedó todo bien y sin diferencias.

Saludos

**Comment by ivanduva** [comment]
Perfecto, el trabajo integrador queda aprobado.
Buen trabajo!
Saludos
